fakebook/src/app/component/newsfeed/newsfeed.component.ts

```diff
@@ -12,22 +13,43 @@ import { PostService } from '../../service/post.service';
 export class NewsfeedComponent implements OnInit {
   posts: Post[] = [];
   user: User | null = null;
+  postId: number;
 
   constructor(
     private newsfeedService: NewsfeedService,
-    private postService: PostService
-  ) {}
+    private postService: PostService,
+    private route: ActivatedRoute
+  ) {
+    this.route.params.subscribe(params => {
+      console.log(params);
+      if(params['id'] != undefined){   
+        this.postId = +params['id'];
+          if(this.postId != undefined)
+            this.getPostsById();
+      }
+      else{
+        this.getPosts();
+      }
```

Comment:
fix formatting, use strict equals, and consider what will happen if someone types in "newsfeed/asdf": postId will become NaN and we'll send NaN to the server.

Reply:
Fixed.